templates/problem/voting-controls.html

```diff
@@ -0,0 +1,218 @@
+    }
+
+    .vote-form-text {
+        font-size: 18px;
+    }
+
+    .voting-form-error {
+        font-size: 18px;
+        color: red;
+    }
+
+    .vote-button {
+        float:right;
+    }
+
+    .delete-vote-button{
+
+    }
+
+    .form-button {
+        cursor: pointer;
+    }
+</style>
+
+{% if not request.in_contest %}
+    <hr style="padding-top: 0.1em">
+{% endif %}
+
+{% include 'lightbox.html' %}
+
+{% if can_vote %}
+    <div class="vote-form" id="id_vote_form_box">
+        <div id="id_has_voted_prompt">
+            <span class="vote-form-info">{{ _('Current Vote:') }}</span>
+            <span id="id_current_vote_value" class="vote-form-value"></span>
+            <br/>
+            <br/>
+            <span class="vote-form-text">{{  _('Change your vote below!') }}</span>
+            <br/>
+            <br/>
+        </div>
+        <div id="id_has_not_voted_prompt">
+            <span class="vote-form-text">{{ _("Cast your vote on this problem's weighting below!") }}</span>
+            <br/>
+            <br/>
+        </div>
+        <form id="id_delete_vote_form">
+            {% csrf_token %}
+        </form>
+        <form id="id_vote_form">
+            {% csrf_token %}
+            <span class="vote-form-info">{{ _('Points') }}</span>
+            <input id="id_points" class="vote-form-value" type="number" step="1"
+                   min="{{ min_possible_vote }}" max="{{ max_possible_vote }}" name="points">
+            <br/>
+            <div id="id_points_error_box">
+                <br/>
+                <span id="id_points_error" class="voting-form-error"></span>
+                <br/>
+            </div>
+            <br/>
+            <span class="vote-form-info">{{ _('Note') }}</span>
+            <br/>
+            <textarea id="id_note" class="vote-form-text" name="note" rows="6" cols="80"></textarea>
+            <br/>
+            <div id="id_note_error_box">
+                <br/>
+                <span id="id_note_error" class="voting-form-error"></span>
+                <br/>
+            </div>
+            <br/>
+            <div>
+                <input type="submit" id="id_vote_form_submit_button" class="vote-button" value="{{ _('Vote') }}">
+                <input type="submit" id="id_delete_vote_form_submit_button" class="delete-vote-button" value="{{ _('Delete vote') }}" form="id_delete_vote_form">
+            </div>
+        </form>
+    </div>
+
+    <div>
+        <a class="form-button" id="id_vote_button"></a>
+    </div>
+
+    {% if vote is not none %}
+        {{ vote.note|json_script('voted_note') }}
+    {% endif %}
+    <script>
+        let voted_points = null;
+        let voted_note = null;
+        {% if vote is not none %}
+            let has_voted = true;
+            voted_points = {{ vote.points }};
+            voted_note = JSON.parse(document.getElementById('voted_note').textContent);
+        {% else %}
+            let has_voted = false;
+        {% endif %}
+
```

Comment:
Should we get this data in some other way? constructing JS code with templates hurts me

Reply:
uh how else?